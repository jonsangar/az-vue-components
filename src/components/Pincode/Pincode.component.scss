.pincode,
.pincode__mask{
  top: 0;
  left: 0;
  z-index: 100;
  width: 100%;
  height: 100%;

}

.pincode {
  position: fixed;
  display: flex;
  align-items: center;
  justify-items: center;
  color: color(gray7);

  &__mask {
    position: absolute;
    background: rgba(0,0,0,.9);
  }

  &__wrapper{
    position: relative;
    z-index: 101;
    width: 100%;
    overflow: hidden;

    .pincode-input {
      display: flex;
      justify-content: space-evenly;
      margin-bottom: double();

      input {
        width: 3rem;
        padding: 1rem;
        font-size: font-size(h4);
        color: color(gray7);
        text-align: center;
        background-color: transparent;
        border: 1px solid color(gray6);
        border-radius: $border-radius;
      }


      &__digitÂ {
        margin-right: 2px;
      }
    }
  }

  .row{
    display: flex;
    justify-content: space-evenly;
    margin-bottom: simple();
  }

  .pincode-keyboard{


    &__key{
      padding: 1rem 1.5rem;
      margin: 0;
      font-size: font-size(h5);
      text-align: center;
    }
  }



}
