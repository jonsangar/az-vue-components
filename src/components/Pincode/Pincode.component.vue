<template lang="pug">
.pincode
  .pincode__mask

  .pincode__wrapper
    .pincode-input
      .pincode-input__digit
        input(
          type='number'
          min="0"
          max="9")
      .pincode-input__digit
        input(
          type='number'
          min="0"
          max="9")
      .pincode-input__digit
        input(
          type='number'
          min="0"
          max="9")
      .pincode-input__digit
        input(
          type='number'
          min="0"
          max="9")

    .pincode-keyboard
      .row
        .pincode-keyboard__key.pincode-keyboard__key--1 1
        .pincode-keyboard__key.pincode-keyboard__key--2 2
        .pincode-keyboard__key.pincode-keyboard__key--3 3
      .row
        .pincode-keyboard__key.pincode-keyboard__key--4 4
        .pincode-keyboard__key.pincode-keyboard__key--5 5
        .pincode-keyboard__key.pincode-keyboard__key--6 6
      .row
        .pincode-keyboard__key.pincode-keyboard__key--7 7
        .pincode-keyboard__key.pincode-keyboard__key--8 8
        .pincode-keyboard__key.pincode-keyboard__key--9 9
      .row
        .pincode-keyboard__key.pincode-keyboard__key--0 0
</template>

<script>
export default {

}
</script>

<style lang="scss" src="./Pincode.component.scss" scoped></style>
