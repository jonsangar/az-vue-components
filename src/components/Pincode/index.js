import Pincode from './Pincode.component.vue';

export default Pincode;
