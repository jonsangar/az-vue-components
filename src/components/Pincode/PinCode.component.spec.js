import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerÃ­as de estilos e iconos
import SvgIcon from 'vue-svgicon'
import Pincode from './Pincode.component.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('PinCode.vue', () => {
  const build = () => {
    const wrapper = shallowMount(Pincode, {
      localVue,
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente Pincode con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()

  })
});
