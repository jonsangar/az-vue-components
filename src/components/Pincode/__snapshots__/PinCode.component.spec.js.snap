// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinCode.vue Comprueba renderizado del componente Pincode con snapshot 1`] = `
<div class="pincode">
  <div class="pincode__mask"></div>
  <div class="pincode__wrapper">
    <div class="pincode-input">
      <div class="pincode-input__digit"><input type="number" min="0" max="9"></div>
      <div class="pincode-input__digit"><input type="number" min="0" max="9"></div>
      <div class="pincode-input__digit"><input type="number" min="0" max="9"></div>
      <div class="pincode-input__digit"><input type="number" min="0" max="9"></div>
    </div>
    <div class="pincode-keyboard">
      <div class="row">
        <div class="pincode-keyboard__key pincode-keyboard__key--1">1</div>
        <div class="pincode-keyboard__key pincode-keyboard__key--2">2</div>
        <div class="pincode-keyboard__key pincode-keyboard__key--3">3</div>
      </div>
      <div class="row">
        <div class="pincode-keyboard__key pincode-keyboard__key--4">4</div>
        <div class="pincode-keyboard__key pincode-keyboard__key--5">5</div>
        <div class="pincode-keyboard__key pincode-keyboard__key--6">6</div>
      </div>
      <div class="row">
        <div class="pincode-keyboard__key pincode-keyboard__key--7">7</div>
        <div class="pincode-keyboard__key pincode-keyboard__key--8">8</div>
        <div class="pincode-keyboard__key pincode-keyboard__key--9">9</div>
      </div>
      <div class="row">
        <div class="pincode-keyboard__key pincode-keyboard__key--0">0</div>
      </div>
    </div>
  </div>
</div>
`;
